core/iwasm/aot/aot_runtime.c

```diff
@@ -1188,8 +1626,14 @@ aot_instantiate(AOTModule *module, AOTModuleInstance *parent,
     if (stack_size == 0)
         stack_size = DEFAULT_WASM_STACK_SIZE;
 #if WASM_ENABLE_SPEC_TEST != 0
-    if (stack_size < 48 * 1024)
-        stack_size = 48 * 1024;
+#if WASM_ENABLE_TAIL_CALL == 0
+    if (stack_size < 128 * 1024)
+        stack_size = 128 * 1024;
+#else
+    /* Some tail-call cases require large operand stack */
```

Comment:
@wenyongh
which cases are they?
tail-call tests consuming much stack sounds like a real bug in our implementation, doesn't it?

Reply:
The cases can be found: https://github.com/WebAssembly/gc/blob/master/test/core/return_call.wast#L105-L118.

Yes, it is really an issue, due to our limited knowledge, we didn't find a better way to improve it when implementing AOT tail call feature. You also mentioned the problem in https://github.com/bytecodealliance/wasm-micro-runtime/issues/2231, do you think we can resolve it with musttail attribute or is there other way to resolve it? Thanks.

Reply:
ok. i commented my current thoughts in https://github.com/bytecodealliance/wasm-micro-runtime/issues/2231.